en/developer/reference/articles/02-workflow/05-notification-templates.markdown

```diff
@@ -0,0 +1,287 @@
+| ${browserSniffer}            |          | &#10004;  | [BrowserSniffer](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/servlet/BrowserSniffer.html) |
+| ${portletProviderAction}     | &#10004; |           | Returns the outcome of the [PortletProvider#Action](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/portlet/PortletProvider.Action.html) call |
+| ${taskName}                  | &#10004; |           | The name of the task the notification is in |
+| ${httpUtil}                  |          | &#10004;  | [HttpUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/HttpUtil.html) |
+| ${portletURLFactory}         |          | &#10004;  | [PortletURLFactory](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/portlet/PortletURLFactory.html) |
+| ${imageToken}                | &#10004; |           | DESCRIBE |
+| ${groupPermission}           |          | &#10004;  | [GroupPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/GroupPermission.html) |
+| ${timeZoneUtil}              |          | &#10004;  | [TimeZoneUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/TimeZoneUtil.html) |
+| ${unicodeLanguageUtil}       |          | &#10004;  | [UnicodeLanguageUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/language/UnicodeLanguageUtil.html) |
+| ${expandoRowLocalService}    |          | &#10004;  | [ExpandoRowLocalService](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/expando/kernel/service/ExpandoRowLocalService.html) |
+| ${auditRouterUtil}           |          | &#10004;  | [AuditRouterUtil](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/audit/AuditRouterUtil.html) |
+| ${kaleoTaskInstanceToken}    |          | &#10004;  | [See the documentation on workflow scripting](/docs/7-2/user/-/knowledge_base/u/leveraging-the-script-engine-in-workflow) |
+| ${accountPermission}         |          | &#10004;  | [AccountPermission](https://docs.liferay.com/dxp/portal/7.2-latest/javadocs/portal-kernel/com/liferay/portal/kernel/service/permission/AccountPermission.html) |
+| ${httpUtilUnsafe}            |          | &#10004;  | NEED HELP, does this access restricted variables somehow? [HttpUtilUnsafe]() |
+| ${workflowTaskAssignees}     | &#10004; |          | [See the documentation on workflow scripting](/docs/7-2/user/-/knowledge_base/u/leveraging-the-script-engine-in-workflow) |
+| ${random}                   |           |           | I DONT KNOW  |
```

Comment:
@rafaprax another one i could use help with :point_up_2:

Reply:
It is a instance of java.util.Random a utility class to generate random numbers, see https://docs.oracle.com/javase/8/docs/api/java/util/Random.html.